concordia/views.py

```diff
@@ -202,59 +216,22 @@ def get_template_names(self):
 class CollectionView(TemplateView):
     template_name = "transcriptions/create.html"
 
+    @transaction.non_atomic_requests
     def post(self, *args, **kwargs):
         context = self.get_context_data()
-        name = self.request.POST.get("name")
-        url = self.request.POST.get("url")
-        result2 = None
-        try:
-            result = download_async_collection.delay(url)
-            result.ready()
-            result.get()
-            result2 = check_completeness.delay()
-            result2.ready()
-            result2.get()
-        except Exception as e:
-            pass
-
-        if result2 and not result2.state == "PENDING":
-
-            base_dir = settings.BASE_DIR
-            collection_path = (
-                settings.MEDIA_ROOT + "/concordia/" + name.replace(" ", "-")
-            )
-            os.system("rm -rf {0}".format(collection_path))
-            os.makedirs(collection_path)
-            cmd = "cp -r {0}/* {1}".format("/concordia_images", collection_path)
-            if os.WEXITSTATUS(os.system(cmd)) == 0:
-                os.system("rm -rf {0}".format("/concordia_images/*"))
-            c = Collection.objects.create(
-                title=name, slug=name.replace(" ", "-"), description=name
-            )
-            count = 0
-            for root, dirs, files in os.walk(collection_path):
-                for filename in files:
-                    filename = os.path.join(root, filename)
-                    if True:
-                        count += 1
-                        title = "{0} asset {1}".format(name, count)
-                        media_url = os.path.join(root, filename).replace(
-                            settings.MEDIA_ROOT, ""
-                        )
-                        Asset.objects.create(
-                            title=title,
-                            slug=title.replace(" ", "-"),
-                            description="{0} description".format(title),
-                            media_url=media_url,
-                            media_type="IMG",
-                            collection=c,
-                        )
-            # os.system('rm -rf {0}'.format('/concordia_images/*'))
-            c.is_active = 1
-            c.save()
-
-            return redirect("/transcribe/" + name.replace(" ", "-"))
-        return render(self.request, self.template_name, {"error": "yes"})
+        name = self.request.POST.get('name')
+        url = self.request.POST.get('url')
+        slug = name.replace(' ', '-')
+        collection_path = os.path.join(settings.MEDIA_ROOT, "concordia", slug)
+        c = Collection.objects.create(title=name, slug=slug, description=name)
+        c.copy_images_to_collection(url, collection_path)
+        c.create_assets_from_filesystem(collection_path)
+        c.is_active=1
```

Comment:
Rather than setting c.is_active=1, we will need a feature for admins to "release" a collection. Not sure if that is already ticketed somewhere. No change needed here immediately, but it will need to change in the next sprint.